README.md

```diff
@@ -56,6 +56,49 @@ make serve
 
 This will start the local Hugo server on port 1313. Open up your browser to http://localhost:1313 to view the website. As you make changes to the source files, Hugo updates the website and forces a browser refresh.
 
+## Building the API reference pages
+
+The API reference pages located in `content/en/docs/reference/kubernetes-api/ref` are built from the Swagger specification, using https://github.com/feloy/kubernetes-resources-reference.
+
+To update the reference pages for a new Kubernetes release (replace v1.20 in the following examples with the release to update to):
+
+1. Pull the `kubernetes-resources-reference` submodule:
+
+```
+git submodule update --init --recursive --depth 1
+```
+
+2. Create a new API revision into the submodule, and add the Swagger specification:
+
+```
+mkdir api-ref-generator/api/v1.20
+curl 'https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json' > api-ref-generator/api/v1.20/swagger.json
```

Comment:
@feloy  What version of the spec are you really pulling in?
Getting the right version of the spec is important, otherwise the generated content is not accurate.
If I recall correctly, the spec generated from k8s/k8s does **not** contain the version number.
Readers will notice that the content is supposed to be 1.19.0 but in reality is 1.19.+, or whatever versions is sitting at master.
There may be an easier way to do this:

```
git clone --depth=1 --single-branch --no-checkout --branch v${K8S_RELEASE} https://github.com/kubernetes/kubernetes.git src/k8s.io/kubernetes
cd src/k8s.io/kubernetes && git show "v$(K8SRELEASE):api/openapi-spec/swagger.json" > swagger.json

```

Reply:
Currently it is the swagger I extracted from the release-1.19 branch.

You are right, the swagger does not contain any information about the version.

What I'm not sure is when we will get the swagger file for the new revision. For example today if we want to prepare the API reference docs for the v1.20 release, the release-1.20 does not exist yet, and that's why I'm specifying the master branch in the Makefile.

I think we first have to decide if we want to prepare the API reference docs before the release is done (and so are using the master branch) or if we wait for the release to be done, and so are using the release branch. What's your opinion?

Reply:
Also, we need to indicate in the documentation what is the release documented. Where can we set this version number?

Reply:
Currently, you can run one script from the website and generate all reference documentation. That said, there are times when the generator configuration files (which apparently are still present) need to be updated manually.
A goal would be to get rid of any manual updates to the generator configuration and figure out a better way to consume updates to the spec from k8s/k8s. Also, the tool will need to be tested and maintained in a controlled (quality) way so that the content produced can be depended upon. I am not convinced that the website is the place to do this.

The current script does **not** update the spec with the version, but instead publishes a string on the page indicating the release version (semver) and the date the page was published. I don't know if this is necessary, but adding a publish string provides a hint to the reader that the page was generated by a tool.
The publish date already appears on the bottom of each page (Git commit).

The API reference can be generated before the release if you provide the latest tagged version.
The latest tagged version is nearly identical to the future versioned spec. In other words, the spec
does not change very much from the latest release candidate version to the next GA version.

Reply:
I looked at the nav again. 🚀 
Once you select a page, I think the page content is clearly spaced. I did not find any examples.
I think you could remove "Resources" from the category directory names. What do you think?

The top of the page duplicates the `apiVersion` from the resource.
You could highlight (format in code style) the "apiVersion" in the resource definition (and `kind` ?).
I think the duplicate resource name (title and level2 heading) is confusing. The main resource for the page happens to
be the root object listed and the resource name corresponds to the title of the page.

Do you want to add a version number? Perhaps add a custom version shortcode/label to the right hand panel before the TOC, or add a small version logo/icon/label by the top resource name (title).

What do you think about adding a "published by the kubernetes-resources-reference" string in the footer if the content-type is `api-reference` and link to the source repository.

Operations:
The right hand nav includes the operation headings, but does not specifically indicate GET, POST, PUT, DELETE. 
Instead there is list, create, patch, delete, deletecollection, update, followed by a short description.

Types:
How would someone find out about a type such as `HTTPIngressRuleValue` or `LoadBalancerIngress`? I did not see these types linked in `Ingress`.